### PR TITLE
add health check and version API

### LINKED ISSUE #58: add health check in the api
The guest/agent may stall on some cases or panic. but the vm look normal out side.
health check can be used to check this situation. In hyperstart runtime call the Version()
api periodically. In kata, we also need such mechanism.

[grpc health check](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) might help.

### LINKED ISSUE #61: add grpc version API
We will need a version mechanism for agent gRPC. The most likely case we will want to handle is runtime upgrading and it reconnects to an old agent. While we can keep gRPC API backward compatibility, runtime needs to know this situation and take care not to send new API requests to old agents.

We can start by having a version check API first. And it can live with the health check service.

### PR DESCRIPTION
**Pull request by bergwolf**
fixes #61  
fixes #58

### CODE REVIEW

File: protocols/grpc/health.proto [hunk 1]
```diff
@@ -21,6 +21,12 @@ message HealthCheckResponse {
 	ServingStatus status = 1;
 }
 
+message VersionCheckResponse {
+	string grpc_version = 1;
+	string impl_version = 2;
```
Comment (sameo) [inline protocols/grpc/health.proto]: Could we please rename that one to `agent_version` ?

Comment (bergwolf) [reply protocols/grpc/health.proto]: done.

**Comment by sameo** [comment]
LGTM

File: agent.go [hunk 2]
```diff
@@ -317,10 +317,12 @@ func (s *sandbox) startGRPC() error {
 
 	grpcImpl := &agentGRPC{
 		sandbox: s,
+		version: version,
 	}
 
 	grpcServer := grpc.NewServer()
 	pb.RegisterAgentServiceServer(grpcServer, grpcImpl)
+	pb.RegisterHealthServer(grpcServer, grpcImpl)
```
Comment (sboeuf) [inline agent.go]: Why is this call needed ? I thought this would be part of the same API.

**Comment by WeiZhang555** [comment]
code LGTM, but hmmm,  codecov reports error again

**Comment by sameo** [comment]
@WeiZhang555 Because the added code is not unit tested. I'll add some gogo generated unit tests, this would look better.
For now, I'm merging that one.

**Pull request merged**